source/transactions/tests/pin-mongos.yml

```diff
@@ -483,10 +483,12 @@ tests:
           errorLabelsContain: ["TransientTransactionError"]
           errorLabelsOmit: ["UnknownTransactionCommitResult"]
       # Session unpins from the first mongos after the insert error and
-      # commitTransaction selects the second mongos and fails. If a driver
-      # mistakenly remains pinned then this commit succeed.
+      # commitTransaction selects the second mongos and fails. If this commit
```

Comment:
Instead of "and fails", say something like "which is unaware of the transaction and therefore fails with NoSuchTransaction error"